cibuildwheel/util.py

```diff
@@ -21,15 +21,18 @@ def get_build_verbosity_extra_flags(level):
         return []
 
 
-class BuildSkipper(object):
-    def __init__(self, skip_config):
-        self.patterns = skip_config.split()
+class BuildSelector(object):
+    def __init__(self, build_config, skip_config):
+        self.build_patterns = build_config.split()
+        self.skip_patterns = skip_config.split()
 
     def __call__(self, build_id):
-        return any(fnmatch(build_id, pattern) for pattern in self.patterns)
+        def match_any(patterns):
+            return any(fnmatch(build_id, pattern) for pattern in patterns)
+        return match_any(self.build_patterns) and not match_any(self.skip_patterns)
```

Comment:
So the logic is 'match any of the build patterns, and none of the skip patterns.' Makes sense!

> (b1 or b2 or b3) and not (s1 or s2 or s3)  
> (b1 or b2 or b3) and not s1 and not s2 and not s3

Reply:
Credit to @altendky for coming up with those semantics. But indeed, I've been thinking a little bit more, and this gives a really intuitive interpretation (there's not even really a 'first `CIBW_BUILD` is applied, and then `CIBW_SKIP`' kind of sequential semantics)